### DISCUSSION TITLE
Is there a way to turn off the node core hyperlink icon for internal links?

### DISCUSSION THREAD

**Discussion by kenfhill84083** [freeplane/freeplane#1256] [Help] [answered]
Is there a way to turn off the hyperlink icon for internal links in the node core?  I thought I remembered doing this in the past but cannot find any info about it now.

back in the old forum [this thread](https://sourceforge.net/p/freeplane/discussion/758437/thread/f522dba717/) indicates it can be done with Vew > View Settings > Show Icons.  But that does not work here.

Any counsel?  Thanks.

**Comment by kenfhill84083** [comment]
> Is there a way to turn off the hyperlink icon for internal links in the node core? I thought I remembered doing this in the past but cannot find any info about it now.

clarification:  I mean the kind of link that looks like this : freeplane:/%20/K:/progam%20data/freeplane/data/asdf.mm#ID_381307424

In the node core it has a red triangle and the Freeplane logo.

![freeplane _ local link icon](https://github.com/freeplane/freeplane/assets/6380301/06e5f695-af94-473c-a4e1-2b703049ad77)

**Reply by EdoFro** [reply]
It's a link to a specific node in another map. Surely the parent map from where you exported a branch as the new map.

Do you need that link?

Do you need that that link points to an specific node?

just modify the link, not?

**Comment by kenfhill84083** [comment]
> It's a link to a specific node in another map. Surely the parent map from where you exported a branch as the new map.
> 
> Do you need that link?
> 
> Do you need that that link points to an specific node?
> 
> just modify the link, not?

Unfortunately, I took a break from Freeplane for a few years and I am still having to relearn stuff (including basic stuff).  Hard for an old brain. 

I have been making links within a map using "Edit->Copy->Copy node URI" and pasting that in as the link.  I can see now I should have been using the anchor method for what I wanted to accomplish.  I will probably have to go back and fix a bunch of links.  (Good thing I can open them in another map view - that should help me a lot.)

But either way, sometimes it would still be useful for me to hide the link icon for either type of link.  Or at least make them less obtrusive.   Like when I want to take a screenshot of some part of the map.

**Reply by EdoFro** [reply]
the only way I can think of, is to add your own svg icon in the user directory that has a less obstrusive look. Freeplane would use that instead of the standard one.

**Reply by kenfhill84083** [reply]
Do you happen to know the filename of that icon?

**Reply by EdoFro** [reply, marked as answer]
![image](https://github.com/freeplane/freeplane/assets/35700575/cfaf13ef-336e-4913-91da-a23462ee5a1d)

I would try these names.

**Reply by EdoFro** [reply]
I think the original files are here:
https://github.com/freeplane/freeplane/tree/1.11.x/freeplane/src/viewer/resources/images/icons/links

**Reply by EdoFro** [reply]
and put the modified files in `<userdirectory>/icons/links/` and restart Freeplane.

Maybe there is a better way, but I don't know.

**Reply by macmarrum** [reply]
> I have been making links within a map using "Edit->Copy->Copy node URI" and pasting that in as the link. I can see now I should have been using the anchor method for what I wanted to accomplish. I will probably have to go back and fix a bunch of links.

Here's a script that converts such Node-URI links to simple Local Links in the entire mind map.
As always, create a backup of your mind map first :smile: 

[findLocalLinksWithFullPathAndConvertToLocal.groovy](https://raw.githubusercontent.com/macmarrum/freeplane-scripting/main/src/scripts/findLocalLinksWithFullPathAndConvertToLocal.groovy)

**Reply by kenfhill84083** [reply]
That is amazing.  Is "Coder" your middle name?  Thanks.  Works great. This should save some time (although I notice I have some other problems I am not going to detail because (1) they are manageable and (2) I don't want to seem like I'm hinting for more scripts). :)

**Comment by kenfhill84083** [comment]
Thanks again.

**Comment by amariusz** [comment]
> I have been making links within a map using "Edit->Copy->Copy node URI" and pasting that in as the link. I can see now I should have been using the anchor method for what I wanted to accomplish. I will probably have to go back and fix a bunch of links. (Good thing I can open them in another map view - that should help me a lot.)

Another way to quicky link 2 nodes within the same map - select source node, select target node, press Alt+Shift+L (add local hyperlink).
